judge/models/problem.py

```diff
@@ -451,6 +451,32 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def solved_by(self, user):
+        # Return true if a full AC submission to the problem from the user exists.
+        return self.submission_set.filter(user=user.profile, result='AC', points=F('problem__points')).exists()
+
+    # 0 = nothing, 1 = view only, 2 = edit and view
+    def get_vote_perm(self, user):
```

Comment:
We should probably use an enum, like #1886.

Reply:
seems tedious
i'll leave this to another reviewer

Reply:
I think an enum would be nice. It could probably also be used in other places which have a similar no perm/view perm/edit perm structure.

Reply:
How do we want to split this up? We could push this code and refactor to an enum (along with other places) later, or we could refactor just this part to an enum and refactor the rest of the places later, or we could refactor everything now (I don't like this option).

Reply:
done, see `class VotePermission(IntEnum)`